vignettes/GeospatialDataIntegration.Rmd

```diff
@@ -0,0 +1,647 @@
+assessments and reporting water quality conditions to EPA and the
+public.
+
+**Intended Audience**
+
+Water Quality eXchange (WQX) and Water Quality Portal (WQP) community,
+Clean Water Act (CWA) community (EPA, States and Tribal Nations), water
+quality and geospatial data analysts/researchers, EPA/USGS and other
+federal agencies.
+
+**Leveraged R Packages**
+
+-   EPA: EPATADA, StreamCatTools, spsurvey, spmodel, SSN2
+
+-   USGS: dataRetrieval, nhdplusTools, hydroloom
+
+-   Fundamental geospatial packages: sf, prism, terra, leaflet and tmap
+
+# Install and load packages
+
+We will be leveraging the EPATADA R Package for WQP data retrieval,
+cleaning, visualization and other steps needed to prepare for analysis.
+Let's dive into Green Bay, WI!
+
+First, install and load the remotes package specifying the repo. This is
+needed before installing EPATADA because it is only available on GitHub
+(not CRAN).
+
+```{r install_remotes, results = 'hide'}
+install.packages("remotes",
+  repos = "http://cran.us.r-project.org"
+)
+library(remotes)
+```
+
+Next, install and load EPATADA using the remotes package. All other
+dependencies for this workflow will be downloaded automatically.
+
+```{r devsetup, include = F, eval = T}
+remotes::install_github("USEPA/EPATADA", 
+                        ref = "geospatial-vignette",
+                        dependencies = TRUE,
+                        force = TRUE)
+library(EPATADA)
+```
+
+```{r usersetup, eval = F, results = 'hide'}
+remotes::install_github("USEPA/EPATADA", 
+                        ref = "develop",
+                        dependencies = TRUE,
+                        force = TRUE)
+library(EPATADA)
+```
+
+It's go time! Let's time our process.
+
+```{r startime}
+# Record start time
+start.time <- Sys.time()
+```
+
+# WQP data discovery and cleaning
+
+This is an abbreviated introduction to key TADA Module 1 WQP Data
+Discovery and Cleaning functions. Additional functions and a more
+detailed example workflow is available
+[here](https://usepa.github.io/EPATADA/articles/TADAModule1.html).
+
+**Retrieve data from the WQP**
+
+In this example, we will first use EPA's How's My Waterway (HMW)
+application to find an applicable Hydrologic Unit Code (HUC) for our
+area of interest - the [Fox River, Green Bay,
+WI](https://mywaterway.epa.gov/community/040302040405/monitoring). Next,
+let's query the WQP using the identified HUC, state abbreviation, and a
+date range. In this example, we'll start by pulling all data available
+in the WQP for this HUC 12 in Wisconsin for the last 5 years.
+
+WATERSHED: City of Green Bay-Fox River (040302040405)
+
+```{r getdata}
+# # Uncomment to query the WQP
+# GreenBay_FoxRiver <- TADA_DataRetrieval(
+#   statecode = "WI",
+#   startDate = "2015-01-01",
+#   endDate = "2024-12-30",
+#   huc = c("040302040405"),
+#   applyautoclean = TRUE
+# )
+
+GreenBay_FoxRiver <- NMCWorkshopData::GreenBay_FoxRiver
```

Comment:
vignette doesn't show installing NMCWorkshopData data package - should we get rid of that line and uncomment the normal way of pulling data (since data package was a convenience for the workshop in case of no wifi connectivity?)